dashboard/app/controllers/aichat_controller.rb

```diff
@@ -212,4 +236,26 @@ def find_toxicity
     limit = DCDO.get('aichat_request_limit_per_min', DEFAULT_REQUEST_LIMIT_PER_MIN)
     Cdo::Throttle.throttle(AICHAT_PREFIX + id.to_s, limit, 60)
   end
+
+  private def can_view_student_chat_history?(student_user_id, level_id, script_id, script_level_id = nil)
```

Comment:
Not blocking, but this reminds me of this discussion thread of this controller being a combination of what should be separated into controllers for `AichatEvent` and `AichatRequest`:

https://codedotorg.slack.com/archives/C06UT1E2796/p1725664712951719

I think this should probably eventually live in `ability.rb` (nice that it is extracted now!), and we can pull out `submit_teacher_feedback` into an event-specific controller)

Reply:
Good call! I'll note this as another follow-up.